src/data/nat/choose/sum.lean

```diff
@@ -59,6 +67,31 @@ begin
        mul_zero, add_zero, pow_succ] }
 end
 
+/-- A version of `commute.add_pow` but with the binomial coefficient applied via scalar action
+of ℕ. -/
+lemma add_pow' :
+  (x + y) ^ n = ∑ m in range (n + 1), choose n m • (x ^ m * y ^ (n - m)) :=
+by simp_rw [_root_.nsmul_eq_mul, cast_comm, h.add_pow n]
+
+/-- A version of `commute.add_pow'` but using `finsum` rather than `finset.sum`. -/
+lemma add_pow'' :
+  (x + y) ^ n = ∑ᶠ m ≤ n, choose n m • (x ^ m * y ^ (n - m)) :=
```

Comment:
Another option:
```suggestion
  (x + y) ^ n = ∑ᶠ m p (h : m + p = n) , choose n m • (x ^ m * y ^ p) :=
```

Reply:
Oh I like this, let me see about a proof.

Reply:
Probably that's worth having as a lemma about `nat_antidiagonal` anyway.

Reply:
I've added proof of this now. This branch is now a bit of a mess but I'll wait till #7439 is resolved till I tidy it up.